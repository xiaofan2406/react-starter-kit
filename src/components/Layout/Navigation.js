import React from 'react';
import PropTypes from 'prop-types';
import withCss from 'react-jss';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'configs';
import { theme, variables } from 'styles';

const css = {
  Navigation: {},
  link: {
    color: theme.inverseColor,
    textDecoration: 'none',
    padding: '0 0.5em',
    display: 'inline-block',
    lineHeight: '42px',
    height: variables.Layout.headerHeight,
    '&:hover': {
      backgroundColor: theme.bgAccentColor
    }
  },
  linkActive: {
    borderBottom: `2px solid ${theme.primaryColor}`
  }
};

function Navigation({ classes }) {
  return (
    <div className={classes.Navigation}>
      {Object.values(ROUTES).map(route =>
        <NavLink
          className={classes.link}
          activeClassName={classes.linkActive}
          key={route.path}
          exact={route.exact}
          to={route.path}
        >
          {route.name}
        </NavLink>
      )}
    </div>
  );
}

Navigation.propTypes = {
  classes: PropTypes.object.isRequired
};

export { Navigation as Component };

export default withCss(css)(Navigation);
