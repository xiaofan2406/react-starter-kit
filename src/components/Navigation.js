/* @flow */
import React from 'react';
import { css } from 'react-emotion';
import { NavLink } from 'react-router-dom';
import { theme, spacing } from 'styles';
import { NAV_LINKS } from 'utils/constants';

const cssNavigation = css`
  position: fixed;
  left: ${spacing.unit}px;
  bottom: ${spacing.unit}px;
  display: flex;
  flex-direction: column;
  border: 1px solid ${theme.borderColor};
  & > .link {
    text-decoration: none;
    padding: 0.5em;
    display: inline-block;
    &:hover {
      background-color: ${theme.borderColor};
    }
    &.active {
      border-right: 2px solid ${theme.primaryColor};
    }
  }
`;

const Navigation = () => (
  <div className={cssNavigation}>
    {Object.keys(NAV_LINKS).map(routeName => (
      <NavLink
        className="link"
        activeClassName="active"
        key={NAV_LINKS[routeName].to}
        exact={NAV_LINKS[routeName].exact}
        to={NAV_LINKS[routeName].to}
      >
        {NAV_LINKS[routeName].name}
      </NavLink>
    ))}
  </div>
);

export { Navigation as Component };

export default Navigation;
