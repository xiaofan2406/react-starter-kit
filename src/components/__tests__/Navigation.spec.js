import React from 'react';
import { render } from '@testing-library/react';
import { expect, test } from '@jest/globals';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../Navigation';

test('Navigation component matches snapshot', () => {
  const { container } = render(
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>
  );

  expect(container.firstChild).toMatchSnapshot();
});
