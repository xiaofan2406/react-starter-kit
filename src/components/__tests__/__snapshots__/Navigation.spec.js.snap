// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
.emotion-0 {
  position: fixed;
  left: 6px;
  bottom: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #f5f5f5;
}

.emotion-0 > .link {
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.5em;
  display: inline-block;
}

.emotion-0 > .link:hover {
  background-color: #f5f5f5;
}

.emotion-0 > .link.active {
  border-right: 2px solid #00bcd4;
}

<div
  className="emotion-0"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="link"
    exact={true}
    key="/"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="link"
    key="/about"
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="link"
    key="/contact"
    to="/contact"
  >
    Contact
  </NavLink>
</div>
`;
